### ISSUE TITLE
Automatic compilation of ebooks and PDF

### ISSUE THREAD

**Issue by jnavila** [progit/progit2-uk#185]
Hi all,

I'm in the process of making an automatic publication of the book on git-scm.com, using asciidoctor and Travis-CI for generating the PDF and epub.

This rework requires to move files around, change how cross-refs are done and change build script. This is quite disruptive and would cause conflicts with ongoing changes.

Please ping me when the repo is idle, so that I can make the transition before you guys restart your work.

Thanks,

**Comment by hedrok** [comment]
Hi,

I was going to update translation with changes of English original for a long time (last update was in April 2016).
So I'll merge progit2/master to progit2-uk/master (and will resolve any merge conflicts by translating what is needed). If your rework is already in progit2/master,  we will get it here as a bonus.

I'll finish this merge during a week and will write when I finish the update.

Thank you.

**Comment by hedrok** [comment]
Unfortunately there are much more conflicts (updates to original) than I anticipated. I'm slowly working on resolving them, in the worst case everything will be ready by 9th of April.

**Comment by jnavila** [comment]
Up to you. Maybe the epubcheck utility that is used for the automatic compilation can help ensure that the book is not broken.

**Comment by jnavila** [comment]
piing @hedrok What's the status?

**Comment by hedrok** [comment]
@jnavila Sorry for long reply - somehow I've missed the email :(
And sorry for missed deadline.
Current status: I've got 88 files with conflicts, 46 of them are resolved... But I haven't had much time - I'll try my best to resolve everthing till 30th of April.
If I fail, I'll ask you to make transition yourself without this merge, but I really hope that won't be needed.

**Comment by hedrok** [comment]
Currently 13 files with conflicts left. Most of today's time I'll work on this project and will try to finish by tomorrow.

**Comment by jnavila** [comment]
OK. thanks for the status.

**Comment by hedrok** [comment]
Nope, 6 files to go, including C-git-commands which I surely will have to automate (113 conflicts).
Unfortunately I won't have a lot of time till Saturday, but I'll try to finish everything during this week.

**Comment by hedrok** [comment]
Merged at last.


Checked with epubcheck, got only warnings because of non-ascii filenames:

```
Validating using EPUB version 3.0.1 rules.
WARNING(PKG-012): ../../progit.epub/OEBPS/_автори.xhtml(-1,-1): File name contains the following non-ascii characters: автори. Consider changing the filename.
WARNING(PKG-012): ../../progit.epub/OEBPS/_вступ.xhtml(-1,-1): File name contains the following non-ascii characters: вступ. Consider changing the filename.
WARNING(PKG-012): ../../progit.epub/OEBPS/_передмова_від_бена_страуба.xhtml(-1,-1): File name contains the following non-ascii characters: передмовавідбенастрауба. Consider changing the filename.
WARNING(PKG-012): ../../progit.epub/OEBPS/_передмова_від_скота_чакона.xhtml(-1,-1): File name contains the following non-ascii characters: передмовавідскотачакона. Consider changing the filename.
WARNING(PKG-012): ../../progit.epub/OEBPS/_присвячується.xhtml(-1,-1): File name contains the following non-ascii characters: присвячується. Consider changing the filename.

Check finished with warnings
```

Also there are currently 4 new non-translated files. I'll translate them later.

Is there something else I should do?

**Comment by jnavila** [comment]
The warnings are safe, this is a "feature" from asciidoctor. If you don't get errors, that is good news that the structure of the book is correct (title level inclusion, cross-references). You already did a lot of the transition by merging the upstream changes. So from this, I'm going to set up the automatic compilation on travis-ci, with a PR.

**Comment by hedrok** [comment]
Ok, thank you.

**Comment by hedrok** [comment]
Fixed by #186
